cmake/quantlib.cmake

```diff
@@ -0,0 +1,90 @@
+cmake_minimum_required(VERSION 3.0)  # Required for CMAKE_VS_PLATFORM_TOOLSET
+
+macro(get_quantlib_library_name QL_OUTPUT_NAME)
+    message(STATUS "QuantLib library name tokens:")
+
+    # MSVC: Give QuantLib built library different names following code in 'ql/autolink.hpp'
+    if(MSVC)
+        
+        # - toolset
+        # ...taken from FindBoost.cmake
+        if (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 19.10)
+            set(QL_LIB_TOOLSET "-vc141")
+        elseif(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 19)
+            set(QL_LIB_TOOLSET "-vc140")
+        elseif(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 18)
+            set(QL_LIB_TOOLSET "-vc120")
+        elseif(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 17)
+            set(QL_LIB_TOOLSET "-vc110")
+        elseif(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 16)
+            set(QL_LIB_TOOLSET "-vc100")
+        elseif(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 15)
+            set(QL_LIB_TOOLSET "-vc90")
+        else()
+            message(FATAL_ERROR "Compiler below vc90 is not supported")
+        endif()
+        message(STATUS " - Toolset: ${QL_LIB_TOOLSET}")
+        
+        # - platform
+        if("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
+            set(QL_LIB_PLATFORM "-x64")
+        endif()
+        message(STATUS " - Platform: ${QL_LIB_PLATFORM}")
+
+        # - thread linkage
+        set(QL_LIB_THREAD_OPT "-mt")  # _MT is defined for /MD and /MT runtimes (https://docs.microsoft.com/es-es/cpp/build/reference/md-mt-ld-use-run-time-library)
+        message(STATUS " - Thread opt: ${QL_LIB_THREAD_OPT}")
+        
+        # - static/dynamic linkage
+        if(${MSVC_RUNTIME} STREQUAL "static")
+            set(QL_LIB_RT_OPT "-s")
+        endif()
+        message(STATUS " - Linkage opt: ${QL_LIB_RT_OPT}")
+        
+        # - release/debug is selected during build, so it cannot be part of the generated name
+        set(CMAKE_DEBUG_POSTFIX "-gd")
```

Comment:
I think that if you move this into the `if(${MSVC_RUNTIME} STREQUAL "static")` above and set it to `"-gd"` for non static and just `"gd"` for static, we'll get the original names.  What do you think?

Reply:
yes, it will work, just a little code duplicity, what we can live with it :+1: